src/renderer/store/modules/utils.js

```diff
@@ -175,6 +175,90 @@ const actions = {
     }
   },
 
+  async downloadMedia({ rootState, dispatch }, [url, title, extension, folder]) {
```

Comment:
I think passing an object is better than an array (as you see for `{ rootState, dispatch }`)
```suggestion
  async downloadMedia({ rootState, dispatch }, { url, title, extension, folder }) {
```